File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        
        HashSet<Integer> hs = new HashSet<Integer>();
        int len = elements.length;
        for(int i = 0; i < len; i++) {
        	int sum = 0;
        	for(int j = 0; j < len; j++) {
        		sum += elements[(i + j) % len];
        		hs.add(sum);
        	}
        }
        
        answer = hs.size();
        
        return answer;
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements : list):
    extended = elements * 2

    size = len(elements)

    sums_set = set()

    for i in range(1, size + 1):
        sums_set.update({sum(extended[x:x+i]) for x in range(0, size)})

    return len(sums_set)